utils/src/main/scala/com/salesforce/op/utils/spark/OpSparkListener.scala

```diff
@@ -197,7 +202,46 @@ case class StageMetrics private
   shuffleWriteTime: Long,
   shuffleBytesWritten: Long,
   shuffleRecordsWritten: Long
-) extends JsonLike
+) extends JsonLike {
+  def plus(sm: StageMetrics): StageMetrics = {
```

Comment:
instead of defining + you can use algebird macro for case classes, i.e.

```scala
import com.twitter.algebird.Operators._
import com.twitter.algebird.macros.caseclass

implicit val stageSG = caseclass.semigroup[StageMetrics]
val sumStages = stage1 + stage2
```